clients/rust/src/hooked/pool.rs

```diff
@@ -77,7 +77,12 @@ impl Pool {
                     Direction::Down => base.checked_div(&factor),
                 };
 
-                let imprecise = result
+                let rounded_result = match side {
+                    TakerSide::Buy => result.unwrap().ceiling(),
```

Comment:
Can we early return these errors instead of unwrapping? Use `?` and if needed you can do `ok_or` to map Option to an error.